activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -732,11 +732,31 @@ def raw_execute(sql, name, async: false, allow_retry: false, uses_transaction: t
           log(sql, name, async: async) do
             with_raw_connection(allow_retry: allow_retry, uses_transaction: uses_transaction) do |conn|
               sync_timezone_changes(conn)
-              conn.query(sql)
+              result = conn.query(sql)
+              handle_warnings(sql)
+              result
             end
           end
         end
 
+        def handle_warnings(sql)
+          return if @raw_connection.warning_count == 0
+
+          result = @raw_connection.query("SHOW WARNINGS")
```

Comment:
I think running this query will also break [`affected_rows`](https://github.com/rails/rails/blob/13783f36ad6c4adbe2121272bd0c4d239bb153ab/activerecord/lib/active_record/connection_adapters/mysql/database_statements.rb#L68) when a warning was generated

Reply:
Yep, we have some special handling for that in `pedant-mysql2`: https://github.com/Shopify/activerecord-pedantmysql2-adapter/blob/de2178a2b60dcf16962791fc72cb16f79ba76bb7/lib/active_record/connection_adapters/pedant_mysql2_adapter.rb#L84

Reply:
Ah, yes will make sure that gets in there 🙇‍♀️